src/include/duckdb/optimizer/join_order/cost_model.hpp

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/optimizer/join_order/cost_model.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "duckdb/optimizer/join_order/join_node.hpp"
+#include "duckdb/optimizer/join_order/cardinality_estimator.hpp"
+
+namespace duckdb {
+
+class CostModel {
+public:
+	CostModel(QueryGraphManager &query_graph_manager);
+
+private:
+	//! query graph storing relation manager information
+	QueryGraphManager &query_graph_manager;
+
+public:
+	void InitCostModel();
+
+	//! Compute cost of a join relation set
+	double ComputeCost(optional_ptr<JoinNode> left, optional_ptr<JoinNode> right);
```

Comment:
Can these be references to `JoinNode`s instead? Seems like in the implementation file that they are always there, so:
```c++
double ComputeCost(JoinNode &left, JoinNode &right);
```